pkg/streamingpromql/testdata/ours/functions.test

```diff
@@ -460,3 +460,38 @@ load 6m
 
 eval range from 0 to 96m step 6m round(series, scalar(toNearest))
   {a="b"}  -5   2.7  0.5   20    10    3.25    0         1000.01 0          -1000000  NaN NaN NaN NaN _ _ 10
+
+# Testing deriv
```

Comment:
[nit] Similar comment to https://github.com/grafana/mimir/pull/9859#discussion_r1839004339 applies here as well

Reply:
Refactored the test in 6fdb1f69ef738f595ba3d2b141afa0f85f65665c